src/Illuminate/Foundation/Console/RouteCacheCommand.php

```diff
@@ -104,6 +104,10 @@ protected function buildRouteCacheFile(RouteCollection $routes)
     {
         $stub = $this->files->get(__DIR__.'/stubs/routes.stub');
 
-        return str_replace('{{routes}}', base64_encode(serialize($routes)), $stub);
+        return str_replace(
+            '{{routes}}',
+            str_replace(['\\', '\'', ], ['\\\\','\\\''], serialize($routes)),
```

Comment:
So why do you have to replace backslashes here, but when doing it in reverse you dont need to add them back in?

Reply:
@laurencei I am afraid about different string representation. Since I usually use double quotes to write a string in the majority of my PHP codes (used for chars that are written in hexadecimal and octal). But, when I read the Laravel source code, it often uses a single quote instead. So I try to follow it.

I just tried to change it to `str_replace('\'', '\\\'', serialize($routes))` in my local. It looks fine so far. I will commit the change of it soon if my benchmark is approved.